sys/gnu.cmake

```diff
@@ -0,0 +1,79 @@
+#
+# Fortran compiler settings
+#
+if(WITH_MPI)
+  set(CMAKE_Fortran_COMPILER "mpifort" CACHE STRING "Fortran compiler")
+else()
+  set(CMAKE_Fortran_COMPILER "gfortran" CACHE STRING "Fortran compiler")
+endif()
+
+set(CMAKE_Fortran_FLAGS "" CACHE STRING "General Fortran flags")
+
+set(CMAKE_Fortran_FLAGS_RELEASE "-O2 -funroll-all-loops" CACHE STRING
+  "Specific Fortran flags for Release (production) mode")
+
+set(FYPP_FLAGS "" CACHE STRING "Flags for the preprocessor")
+
+
+#
+# C compiler settings
+#
+set(CMAKE_C_COMPILER "gcc" CACHE STRING "C compiler")
+
+set(CMAKE_C_FLAGS "" CACHE STRING "General C flags")
+
+set(CMAKE_C_FLAGS_RELEASE "-O2 -funroll-all-loops" CACHE STRING "Specific C flags for Release mode")
+
+
+#
+# External libraries
+#
+
+# LAPACK and BLAS
+if(WITH_MPI)
+  set(LAPACK_LIBRARIES "lapack blas" CACHE STRING "LAPACK and BLAS libraries to link")
+else()
+  set(LAPACK_LIBRARIES "openblas" CACHE STRING "LAPACK and BLAS libraries to link")
```

Comment:
probably should be "blas lapack", firstly as lapack won't always be picked up and secondly as openblas still has that extra thread(s) issue at runtime. Also as mkl is now available, would some comment about using that be a good idea?

Reply:
IMO, if you link statically, blas should come after lapack in the linking order to enable the linker to resolve the blas-calls in lapack (or we would need a --stat-group --end-group construct.) Currently we always link lapack and let the linker throw it out, if not needed. (True for the make-build as well.)

Yes, probably we should mention the possibility of linking MKL, indeed. I'll add such a line.

Reply:
Just changing it to just "blas" here doesn't pick up the lapack symbols for me with cmake 3.10.2, but agree with the point about "lapack blas" (wrong way around in comment)/